File: Cucumber-Projects/SauceDemo/src/test/java/com/automation/pages/LoginPage.java
package com.automation.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class LoginPage {
    WebDriver driver;

    @FindBy(id = "user-name")
    WebElement userNameTextBox;

    @FindBy(id = "password")
    WebElement passwordTextBox;

    @FindBy(id = "login-button")
    WebElement loginButton;

    @FindBy(xpath = "//h3[@data-test='error']")
    WebElement errorMessage;

    public LoginPage(WebDriver driver) {
        this.driver = driver;
        PageFactory.initElements(driver, this);
    }

    public void enterUsername(String username) {
        userNameTextBox.sendKeys(username);
    }

    public void enterPassword(String password) {
        passwordTextBox.sendKeys(password);
    }

    public void clickLogin() {
        loginButton.click();
    }

    public String getErrorMessage() {
        return errorMessage.getText();
    }
}

File: JavaScript-Cucumber-Projects/src/page_objects/loginPage.js
import { By, until } from 'selenium-webdriver';

class loginPage{
    constructor(driver) {
        this.driver = driver;
        this.userNameInput = By.xpath('//input[@name="username"]');
        this.passwordInput = By.xpath('//input[@name="password"]');
        this.submitButton = By.xpath('//button[@type="submit"]');
    }

    async accessToLoginPage(){
        await this.driver.get('https://opensource-demo.orangehrmlive.com');
    }

    async putUser(userName){
        const userNameInput = await this.driver.wait(until.elementLocated(this.userNameInput), 10000);
        await userNameInput.sendKeys(userName);
    }

    async putPassword(passw){
        const passwordInput = await this.driver.wait(until.elementLocated(this.passwordInput), 10000);
        await passwordInput.sendKeys(passw);
    }

    async clickLogin(){
        const submitButton = await this.driver.wait(until.elementLocated(this.submitButton), 10000);
        await submitButton.click();
    }
}

// module.exports = loginPage;
export default loginPage